static/app/utils/profiling/renderers/flamegraphRenderer.tsx

```diff
@@ -0,0 +1,675 @@
+      this.positions[positionOffset + 10] = x2;
+      this.positions[positionOffset + 11] = y2;
+
+      // @TODO check if we can pack bounds across vertex calls,
+      // we are allocating 6x the amount of memory here
+      const boundsOffset = index * VERTICES * BOUNDS_PER_PASS;
+
+      for (let i = 0; i < VERTICES; i++) {
+        const offset = boundsOffset + i * BOUNDS_PER_PASS;
+
+        this.bounds[offset] = x1;
+        this.bounds[offset + 1] = y1;
+        this.bounds[offset + 2] = frame.start + frame.end;
+        this.bounds[offset + 3] = y2;
+      }
+    }
+  }
+
+  initCanvasContext(): void {
+    if (!this.canvas) {
+      throw new Error('Cannot initialize context from null canvas');
+    }
+    // Setup webgl canvas context
+    this.gl = getContext(this.canvas, 'webgl');
+
+    this.gl.enable(this.gl.BLEND);
+    this.gl.blendFuncSeparate(
+      this.gl.SRC_ALPHA,
+      this.gl.ONE_MINUS_SRC_ALPHA,
+      this.gl.ONE,
+      this.gl.ONE_MINUS_SRC_ALPHA
+    );
+    resizeCanvasToDisplaySize(this.canvas);
+  }
+
+  initShaders(): void {
+    if (!this.gl) {
+      throw new Error('Uninitialized WebGL context');
+    }
+
+    // @ts-ignore
+    this.uniforms = {};
+    // @ts-ignore
+    this.attributes = {};
+
+    const vertexShader = createShader(this.gl, this.gl.VERTEX_SHADER, vertex());
+
+    const fragmentShader = createShader(
+      this.gl,
+      this.gl.FRAGMENT_SHADER,
+      fragment(this.theme)
+    );
+
+    // create program
+    this.program = createProgram(this.gl, vertexShader, fragmentShader);
+
+    const uProjectionMatrix = this.gl.getUniformLocation(this.program, 'u_projection');
+    const uModelMatrix = this.gl.getUniformLocation(this.program, 'u_model');
+    const uIsSearchResult = this.gl.getUniformLocation(
+      this.program,
+      'u_is_search_result'
+    );
+    const uBorderWidth = this.gl.getUniformLocation(this.program, 'u_border_width');
+    const uDrawBorder = this.gl.getUniformLocation(this.program, 'u_draw_border');
+
+    if (!uProjectionMatrix) {
+      throw new Error('Could not locate u_projection in shader');
+    }
+    if (!uModelMatrix) {
+      throw new Error('Could not locate u_model in shader');
+    }
+    if (!uIsSearchResult) {
+      throw new Error('Could not locate u_is_search_result in shader');
+    }
+    if (!uBorderWidth) {
+      throw new Error('Could not locate u_border_width in shader');
+    }
+    if (!uDrawBorder) {
+      throw new Error('Could not locate u_draw_border in shader');
+    }
+
+    this.uniforms.u_projection = uProjectionMatrix;
+    this.uniforms.u_model = uModelMatrix;
+    this.uniforms.u_is_search_result = uIsSearchResult;
+    this.uniforms.u_border_width = uBorderWidth;
+    this.uniforms.u_draw_border = uDrawBorder;
+
+    {
+      const aColorAttributeLocation = this.gl.getAttribLocation(this.program, 'a_color');
+
+      // attributes get data from buffers
+      this.attributes.a_color = aColorAttributeLocation;
```

Comment:
Does this need to check if it returned `-1`?

Reply:
Yup, and all the rest under it too!